plotly_express/_core.py

```diff
@@ -1,9 +1,15 @@
+from _plotly_future_ import renderer_defaults, v4_subplots
```

Comment:
can we think of a not-too-nasty way to have this work just like Plotly.py wherein if a certain future flag is set, it uses this mode?

Reply:
And/or just to make it easier to maintain, are there some structural changes here which aren't specific to v4 features that we should backport to `master` sooner to make it easier to maintain this branch?

Reply:
> can we think of a not-too-nasty way to have this work just like Plotly.py wherein if a certain future flag is set, it uses this mode?

This was my original goal, the changes/refactoring ended up being pretty dispersed through the nested hierarchy of functions so I didn't see an easy way to do it.  I'll take another look this evening and see if If can isolate the changes.

Reply:
I'm still on the fence about whether pre-v4-release versions of PX should turn on the future flags. I kinda think they should :) and then we don't need to worry about it too much.

thoughts?

Reply:
I think the "a bunch of pre-release testing from early adopters" is kind of cool

Reply:
Yeah, if it were easy I'd have plotly_express follow the future_flags.  But since I think it would be pretty intrusive, and more complex to maintain, I'm inclined to turn it on by default.

We could also consider keeping plotly_express in an alpha state until the release of plotly.py v4. This way the opt-in to the new behavior would be to install the alpha version.